lib/node/nodes/gravity.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+var Node = require('../node');
+
+var TYPE = 'gravity';
+
+var PARAMS = {
+    source: Node.PARAM.NODE(Node.GEOMETRY.ANY), // "populated places"
+    target: Node.PARAM.NODE(Node.GEOMETRY.POINT), // "malls"
+    weightColunn: Node.PARAM.STRING(),
+    weightThreshold: Node.PARAM.NULLABLE(Node.PARAM.NUMBER()),
+    popColumn: Node.PARAM.STRING(),
+    maxDistance: Node.PARAM.NUMBER(), // meters
+    targetID: Node.PARAM.NUMBER() // the "mall" to be studied
+};
+
+var Gravity = Node.create(TYPE, PARAMS, {cache: true, version: 1});
+
+module.exports = Gravity;
+module.exports.TYPE = TYPE;
+module.exports.PARAMS = PARAMS;
+
+var gravityTemplate = Node.template([
+    'WITH ',
+    ' raw_s as({{=it.sourceQuery}}),',
+    ' raw_t as({{=it.targetQuery}})',
+    ' s as(',
```

Comment:
I ran the analysis and I've got `error: syntax error at or near "s"`

Here the sql:
```
WITH
 raw_s as(select * from atm_machines limit 2),
 raw_t as(select * from atm_machines limit 2 offset 2)
 s as(
SELECT
 array_agg(cartodb_id::bigint) as id
 array_agg(the_geom) as g,
 array_agg(coalesce(cartodb_id,0)::numeric) as p
 FROM raw_s
),
 t as(
SELECT
 array_agg(cartodb_id::bigint) as id
 array_agg(the_geom) as g,
 array_agg(coalesce(cartodb_id,0)::numeric) as p
 FROM raw_t
)
SELECT
 g.source_id as cartodb_id,
 g.the_geom,
 ST_transform(g.the_geom, 3857) as the_geom_webmercator,
 g.h as patronage_prob,
 g.hpop patronaging_pop
 g.dist as dist_to_target
FROM
 t,
 s,
 CDB_Gravity(t.id, t.g, t.w, s.id, s.g, s.p,
  1, 500, null) g
```
Error in psql:
```
ERROR:  syntax error at or near "s"
LINE 4:  s as(
```
And the analysis:
```
{
    source: {
        type: 'source',
        params: {
            query: 'select * from atm_machines limit 2'
        }
    },
    target: {
        type: 'source',
        params: {
            query: 'select * from atm_machines limit 2 offset 2'
        }
    },
    weightColunn: 'cartodb_id',
    popColumn: 'cartodb_id',
    maxDistance: 500,
    targetID: 1
}
```
I recommend you to implement some acceptance test to check those things.